File: backend/src/main/java/ca/metricalsky/winston/entity/ChannelPropertiesEntity.java
package ca.metricalsky.winston.entity;

import jakarta.persistence.Basic;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "channel_properties")
@Getter
@Setter
public class ChannelPropertiesEntity {

    @Id
    @Column(name = "channel_id")
    private String channelId;

    @Basic(optional = false)
    @Column(name = "archived")
    private boolean archived;
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/entity/channel/ChannelPropertiesEntity.kt
package ca.metricalsky.winston.database.entity.channel

import jakarta.persistence.Basic
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "channel_properties")
class ChannelPropertiesEntity(

    @Id
    @Column(name = "channel_id")
    var channelId: String,

    @Basic(optional = false)
    @Column(name = "archived")
    var archived: Boolean
)
